db/migrate/20181113190507_add_orchestration_stacks.rb

```diff
@@ -0,0 +1,22 @@
+class AddOrchestrationStacks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :orchestration_stacks do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
```

Comment:
we should also have ```parent_orchestration_id```

I think references won't allow you to add it with renamed column, so you'll need to add the column and index manually and:

```
add_foreign_key :orchestration_stacks, :orchestration_stacks, on_delete: :cascade, column: 'parent_orchestration_stack_id'
```

Reply:
Is that for a parent stack or a polymorphic belongs_to resource?

Reply:
that would be for the parent stack

Reply:
I don't see this on MIQ so I'll leave it out and we can revisit in a future PR if we need it